lib/src/DateRangePicker/DateRangePickerViewMobile.tsx

```diff
@@ -39,6 +41,7 @@ export const DateRangePickerViewMobile: React.FC<DesktopDateRangeCalendarProps>
   rightArrowButtonProps,
   rightArrowButtonText,
   rightArrowIcon,
+  renderDay = (_, props) => <DateRangeDay {...props} />,
```

Comment:
It looks like we will need to refactor the render APIs. I think that the correct API should be:

```jsx
  renderDay = (props) => <DateRangeDay {...props} />,
```

As far as I have seen this pattern used in the community, it gets one argument, with the equivalent of the props.

I will open a RFC for that breaking change

Reply:
I strongly believe that we need exactly 2 arguments here. And that's why: 

```jsx
renderDay={(day, selectedDate, DayComponentProps) => {
        const date = makeJSDateObject(day); // skip this step, it is required to support date libs
        const isSelected =
          DayComponentProps.inCurrentMonth && highlightedDays.includes(date.getDate());

        return (
          <Badge
            key={date.toString()}
            overlap="circle"
            badgeContent={isSelected ? '🌚' : undefined}
          >
            <Day {...DayComponentProps} />
          </Badge>
        );
      }}
```

Actually an example of rendering custom day will require access to the day that is rendering. I cannot imagine an example where its not. (maybe only if user need to pass custom style). 

If we will pass the `date` as a prop it will need to spend more time to understand how to access the current date. And also will require some kind of notation

```jsx
(DayProps) => <Day {...DayProps} className={getClassByDay(DayProps.day} /> 
```

```jsx
({ day, ...other }) => <Day day={day} {...other} className={getClassByDay(day} /> 
```

I think this notation is the best

```jsx
(day, DayProps) => <Day {...DayProps} className={getClassByDay(day} /> 
```

Reply:
> P.S. `selectedDays` must be removed #1911

Reply:
I don't understand why `day` and `selectedDate` should come from here. It feels that:

- `day` should already be available as part of `PickersDayProps`: https://github.com/mui-org/material-ui-pickers/blob/8f895d5e438dcc8f7f04837763979a3009d777dc/lib/src/views/Calendar/Day.tsx#L80.
- `selectedDate` should be available in the global scope: https://github.com/mui-org/material-ui-pickers/blob/8f895d5e438dcc8f7f04837763979a3009d777dc/docs/pages/demo/datepicker/ServerRequest.example.jsx#L11

Reply:
I have described in my previous answer. Because accessing `renderDay` without using `day` is mostly useless. 
So if we know that `day` is always required we can simplify accessing it from `DayProps.day` to the first argument.  ([react-day-picker](http://react-day-picker.js.org/api/DayPicker/#renderDay)) as reference.

Reply:
I have looked at the codebase of the Autocomplete, we do have cases like this where we decide to inject extra arguments: https://github.com/mui-org/material-ui/issues/19254#issuecomment-643494821. I wonder if it's the best approach. It feels like we could have a single argument, that it would be simpler.

I really like the simplicity of this variant:

```jsx
(DayProps) => <Day {...DayProps} className={clsx(DayProps.className, getClassByDay(DayProps.day}) /> 
```

Reply:
It makes users go to the docs/console.log the props to understand what is passing to the `props`. When there is only one argument – it is easier.

Reply:
> It makes users go to the docs/console.log the props to understand what is passing to the props

Why? We can document the props of the Day component, and link this same page.

Reply:
Yes, we can, and it is documented. But why developer should look and google for props documentation if he need to just render the same day as we do but do some custom things based on the `day of month`? 

I mean it is not so clear as just simple passing of both `day` and `props`.

Reply:
The idea would be that we link the PickersDay component page, right in the DateRangePicker prop page. Making the information accessible with one click. We could also have a demo with the pattern.

I think that the advantage is about not having to wonder about when and which arguments should be exposed. If we don't go down this route. I would recommend that we force the render prop to never have more than 4 arguments, never.